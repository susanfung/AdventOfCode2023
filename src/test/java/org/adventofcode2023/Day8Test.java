package org.adventofcode2023;

import org.approvaltests.Approvals;
import org.junit.jupiter.api.Test;

import java.util.Arrays;
import java.util.List;

class Day8Test {
    @Test
    void stepsExample1() {
        String instructions = "RL";
        List<String> nodes = Arrays.asList("AAA = (BBB, CCC)", "BBB = (DDD, EEE)", "CCC = (ZZZ, GGG)", "DDD = (DDD, DDD)",
                                           "EEE = (EEE, EEE)", "GGG = (GGG, GGG)", "ZZZ = (ZZZ, ZZZ)");
        Approvals.verify(Day8.steps(instructions, nodes));
    }

    @Test
    void stepsExample2() {
        String instructions = "LLR";
        List<String> nodes = Arrays.asList("AAA = (BBB, BBB)", "BBB = (AAA, ZZZ)", "ZZZ = (ZZZ, ZZZ)");
        Approvals.verify(Day8.steps(instructions, nodes));
    }
}