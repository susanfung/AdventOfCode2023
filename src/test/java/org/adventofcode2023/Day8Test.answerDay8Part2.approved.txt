20685524831999